lib/ansible/modules/cloud/vmware/vmware_host_powermgmt_policy.py

```diff
@@ -0,0 +1,210 @@
+  cluster_name:
+    description:
+    - Name of the cluster from which all host systems will be used.
+    - This is required parameter if C(esxi_hostname) is not specified.
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Set the Power Management Policy of a host system to high-performance
+  vmware_host_powermgmt_policy:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    policy: high-performance
+    validate_certs: no
+  delegate_to: localhost
+
+- name: Set the Power Management Policy of all host systems from cluster to high-performance
+  vmware_host_powermgmt_policy:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: '{{ cluster_name }}'
+    policy: high-performance
+    validate_certs: no
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+result:
+    description: metadata about host system's power policy
+    returned: always
+    type: dict
+    sample: {
+        "esxi01": {
+            "msg": "power policy for host 'esxi01' already configured",
+            "error": "",
+        },
+    }
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import PyVmomi, vmware_argument_spec, wait_for_task, TaskError
+from ansible.module_utils._text import to_native
+
+
+class VmwareHostPowerManagement(PyVmomi):
+    """
+    Class to manage power management policy of an ESXi host system
+    """
+    def __init__(self, module):
+        super(VmwareHostPowerManagement, self).__init__(module)
+        cluster_name = self.params.get('cluster_name')
+        esxi_host_name = self.params.get('esxi_hostname')
+        self.hosts = self.get_all_host_objs(cluster_name=cluster_name, esxi_host_name=esxi_host_name)
+        if not self.hosts:
+            self.module.fail_json(msg="Failed to find host system with given configuration.")
+
+    def ensure(self):
+        """
+        Function to manage power management policy of an ESXi host system
+        """
+        results = dict(changed=False, result=dict())
+        policy = self.params.get('policy')
+        host_change_list = []
+        for host in self.hosts:
+            changed = False
+            results['result'][host.name] = dict(msg='', error='')
+
+            if policy == 'high-performance':
+                policy_key = 1
+                policy_short_name = "static"
+            elif policy == 'balanced':
+                policy_key = 2
+                policy_short_name = "dynamic"
+            elif policy == 'low-power':
+                policy_key = 3
+                policy_short_name = "low"
+            elif policy == 'custom':
+                policy_key = 4
+                policy_short_name = "custom"
```

Comment:
this could probably be simplified with using a dict of dicts:
```
power_policies = {
    'high-performance': {
        'key': 1,
        'short_name': 'static'
    },
    'balanced': {
        'key': 2,
        'short_name': 'dynamic'
    },
    'low-power': {
        'key': 3,
        'short_name': 'low'
    },
    'custom': {
        'key': 4,
        'short_name': 'custom'
    }
}
```

Then it is just a matter of using `power_policies[policy]['key']` and `power_policies[policy]['short_name']`

Not really worth a lot of worry over, just a suggestion

Reply:
Makes sense. I will implement that.